test/datasets/test_qnli.py

```diff
@@ -0,0 +1,81 @@
+import os
+import zipfile
+from collections import defaultdict
+from unittest.mock import patch
+
+from parameterized import parameterized
+from torchtext.datasets.qnli import QNLI
+
+from ..common.case_utils import TempDirMixin, zip_equal, get_random_unicode
+from ..common.torchtext_test_case import TorchtextTestCase
+
+
+def _get_mock_dataset(root_dir):
+    """
+    root_dir: directory to the mocked dataset
+    """
+    base_dir = os.path.join(root_dir, "QNLI")
+    temp_dataset_dir = os.path.join(base_dir, "temp_dataset_dir")
+    os.makedirs(temp_dataset_dir, exist_ok=True)
+
+    seed = 1
+    mocked_data = defaultdict(list)
+    for file_name in ("train.tsv", "dev.tsv", "test.tsv"):
+        txt_file = os.path.join(temp_dataset_dir, file_name)
+        with open(txt_file, "w", encoding="utf-8") as f:
+            f.write("index\tquestion\tsentence\tlabel\n")
+            for i in range(5):
+                label = seed % 2
+                rand_string_1 = get_random_unicode(seed)
+                rand_string_2 = get_random_unicode(seed+1)
+                dataset_line = (label, rand_string_1, rand_string_2)
+                label_str = 'entailment' if label == 1 else 'not_entailment'
```

Comment:
Per the convention we are returning integer IDs for the label. Wondering why are return labels in here? On related note, we were thinking of adding some helper info utility that shall provide the label mappings from IDs to original names.

Reply:
We need the return label as a string here because this is the format of the input data file, so in order to write the mock data we cannot rely on integer labels. 

Having label mapping sounds like a good idea, we could have this part of the data loader maybe? But I think it should not be part of this PR, but be a wider change in one separate PR.